backend/src/main/java/org/latexscribe/LatexScribe/webcontroller/TemplateController.java

```diff
@@ -0,0 +1,66 @@
+package org.latexscribe.LatexScribe.webcontroller;
+
+import lombok.RequiredArgsConstructor;
+import org.latexscribe.LatexScribe.domain.dto.TemplateDto;
+import org.latexscribe.LatexScribe.domain.enums.TemplateCategory;
+import org.latexscribe.LatexScribe.domain.model.DocumentTemplate;
+import org.latexscribe.LatexScribe.domain.model.User;
+import org.latexscribe.LatexScribe.service.ITemplateService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/templates")
+public class TemplateController {
+    private final ITemplateService templateService;
+
+    @GetMapping
+    public List<DocumentTemplate> listAll() {
+        return templateService.listAll();
+    }
+
+    @GetMapping("/my_templates")
+    public List<DocumentTemplate> listAllByUser(User user) {
+        return templateService.listAllByUser(user);
+    }
```

Comment:
I think this should be left once we have user APIs e.g. `/api/v1/users/{id}/templates`